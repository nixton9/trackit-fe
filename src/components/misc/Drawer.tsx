import React, { ReactChildren, ReactElement, RefObject } from 'react'
import { Styled } from '../../styles/Drawer.styles'

type DrawerProps = {
  children: ReactChildren | ReactElement
  title: string
  open: boolean
  overlayRef: RefObject<HTMLDivElement>
}

const Drawer: React.FC<DrawerProps> = ({
  children,
  title,
  open,
  overlayRef
}) => (
  <>
    <Styled.DrawerContainer open={open}>
      <Styled.DrawerTitle>{title}</Styled.DrawerTitle>
      <Styled.DrawerContent>{children}</Styled.DrawerContent>
    </Styled.DrawerContainer>
    <Styled.DrawerOverlay ref={overlayRef} open={open} />
  </>
)

export default Drawer
