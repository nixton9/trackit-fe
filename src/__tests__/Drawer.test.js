import React from 'react'
import { render } from '@testing-library/react'
import Drawer from '../components/misc/Drawer'

describe('Drawer', () => {
  it('matches snapshot', () => {
    const open = false
    const overlayEl = null
    const { container } = render(
      <Drawer title="Settings" open={open} overlayRef={overlayEl}>
        <div>
          <h3>First setting</h3>
          <p>Dropdown</p>
        </div>
      </Drawer>
    )

    expect(container).toMatchSnapshot()
  })

  it('shows the title', () => {
    const open = false
    const overlayEl = null
    const { getByText } = render(
      <Drawer title="Settings" open={open} overlayRef={overlayEl}>
        <div>
          <h3>First setting</h3>
          <p>Dropdown</p>
        </div>
      </Drawer>
    )

    expect(getByText('Settings')).toBeInTheDocument()
  })

  it('shows the content of the drawer', () => {
    const open = false
    const overlayEl = null
    const { getByText } = render(
      <Drawer title="Settings" open={open} overlayRef={overlayEl}>
        <div>
          <h3>First setting</h3>
          <p>Dropdown</p>
        </div>
      </Drawer>
    )

    expect(getByText('First setting')).toBeInTheDocument()
    expect(getByText('Dropdown')).toBeInTheDocument()
  })
})
