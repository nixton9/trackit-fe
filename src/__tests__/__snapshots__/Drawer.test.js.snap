// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer matches snapshot 1`] = `
<div>
  <div
    class="Drawerstyles__DrawerContainer-u5jgzg-0 lkcEUU"
  >
    <h2
      class="Drawerstyles__DrawerTitle-u5jgzg-2 eCJKZq"
    >
      Settings
    </h2>
    <div
      class="Drawerstyles__DrawerContent-u5jgzg-3 jAyGXs"
    >
      <div>
        <h3>
          First setting
        </h3>
        <p>
          Dropdown
        </p>
      </div>
    </div>
  </div>
  <div
    class="Drawerstyles__DrawerOverlay-u5jgzg-1 jzRVJQ"
  />
</div>
`;
